### PR TITLE
Enable Dependabot updates to Poetry, fix tests and docs build

### PR DESCRIPTION
**Pull request by d33bs**
This PR focuses on long-term sustainability for this project by enabling Dependabot to automatically update the Poetry lockfile. Normally this is a task which must be performed manually. By enabling this change we can have Dependabot automatically update the dependency references (which still fall into the constraints of the `pyproject.toml` file).

Thanks for any feedback!

Should be merged after #4

### CODE REVIEW

File: docs/src/_static/cosmicqc-example-cytodataframe.png
Comment (jenna-tomkinson) [inline docs/src/_static/cosmicqc-example-cytodataframe.png:1]: This is the exact example I was referencing! That 568 row looks like it should be split into two cells, which I think is the issue with using masks over outlines.

File: tests/data/coSMicQC/show_report/cosmicqc_example_report.png
Comment (jenna-tomkinson) [inline tests/data/coSMicQC/show_report/cosmicqc_example_report.png:1]: Per the discussion at the meeting, should we show coSMicQC outputs in this repo or not? I can see both sides here, so I wanted to open up the conversation here! 😄

File: tests/data/cytotable/NF1_cellpainting_data_shrunken/NF1_plate2_export_masks.cppipe
Comment (jenna-tomkinson) [inline tests/data/cytotable/NF1_cellpainting_data_shrunken/NF1_plate2_export_masks.cppipe:1]: Excited to see this become export outlines! 😄

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! Excited to see the progress of this work!

File: docs/src/_static/cosmicqc-example-cytodataframe.png
Comment (d33bs) [reply docs/src/_static/cosmicqc-example-cytodataframe.png:1]: Thanks, referenced in new issue #8 for further consideration!

File: tests/data/coSMicQC/show_report/cosmicqc_example_report.png
Comment (d33bs) [reply tests/data/coSMicQC/show_report/cosmicqc_example_report.png:1]: Thanks for adding this! I've created a new issue where we can track this: #7 .

File: tests/data/cytotable/NF1_cellpainting_data_shrunken/NF1_plate2_export_masks.cppipe
Comment (d33bs) [reply tests/data/cytotable/NF1_cellpainting_data_shrunken/NF1_plate2_export_masks.cppipe:1]: Thanks! Will continue to track progress here under #8

**Comment by d33bs** [comment]
Thanks @jenna-tomkinson for the review here! I'm going to interpret your review as an approval for #4, as it goes through code which this branch was based on. I'll merge #4 and then this PR to keep the git history clean and descriptive. Related issues were created or referenced to help make sure we address what you mentioned in the comments.

**Comment by d33bs** [comment]
Added changes which are more related to #4 which weren't apparent until after a merge (automated tests were prevented and a docs build failed after merge). I'm trying to keep things succinct by appending the changes here.

**Pull request merged**